python-stdlib/pprint/pprint.py

```diff
@@ -1,6 +1,57 @@
-def pformat(obj):
-    return repr(obj)
+from sys import stdout as _stdout
 
+def pformat(obj, indent=1, sort_dicts=True):
+    'Simply calls pprint() but capturing prints into a string and returning'
+    from io import StringIO as _StringIO
+    stream = _StringIO()
+    pprint(obj, stream=stream, indent=indent, sort_dicts=sort_dicts)
+    return stream.getvalue()
 
-def pprint(obj):
-    print(repr(obj))
+def pp(obj, *args, sort_dicts=False, **kwargs):
+    'Same as pprint() except the default value of sort_dicts is False'
+    pprint(obj, *args, sort_dicts=sort_dicts, **kwargs)
+
+def pprint(obj, stream=None, indent=1, sort_dicts=True):
+    '''
+    Simple implementation of a pretty-printer.
+
+    For simplicity, this does not recurse down.  It does not produce the same
```

Comment:
Would it be possible to make it recurse anyway, as that's much more versatile? Shouldn't be too hard

Reply:
Perhaps not too hard.  The main problem is that the `pprint` function from the standard library has safety checks such as

- make sure not to overwhelm the output stream (by having a maximum width and meticulously checking the output and collapsing it)
- ensure no infinite recursion (such as a list being a member of itself or a descendant of itself)

There are other tricky things like ensuring the indentation is handled properly.  I do not intend to implement those checks at this time as they will add significant complexity.

If you think it would be good to have the option, then I could implement it to have a default of no recursion, but the user can specify a desired depth, at their own peril (if they're not careful) ;)

Reply:
I wouldn't care about the checks etc, but I mainly use pprint for looking at JSON data on the REPL (instead of e.g. opening in a text editor and pretty printing there) and I encounter nested data way more often than not. But that is just my experience.

Reply:
Okay, I've refactored to include depth recursion.  However, since safety checks are not performed, I set the default to a depth of one instead of infinite depth.